client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/useContactQuestioning.ts

```diff
@@ -236,16 +220,59 @@ const useContactQuestioning = (parameters: useContactQuestioningParameters): use
         return (new Set(allIdentificationNumbersToCheck)).size === allIdentificationNumbersToCheck.length;
     }
 
+    
+    const onSubmit = (data: any) => {
+        const ParsedFormData = parseFormBeforeSending(data);
+        saveContactQuestioning(ParsedFormData);
+    };
+
+    const parseFormBeforeSending = (data: any) => {
+        let { form } = data;
+        form.forEach((person: any, index: number) => {
+            form[index] = parsePerson(person, index);
+        });
+
+        return form;
+    };
+
+    /* NOTICE : this is because the form has all of the data that can be edited
+     *          and the server expects all data to be send (like first and last name)
+     *          I'm parsing the data because I'm uncertain why those fields are being sent
+     *          in the first place , if you know why please contact me :)
```

Comment:
Writing comments which are referenced to a specific person is a bad practice considering it won't always be possible to know who wrote the code and even if we could there might be no way of contacting said person.

Reply:
fixed [here](https://github.com/CoronaTeamMOD/CoronaInvestigation/pull/964/commits/eafe44871db81260f0d6ff6de9828e29db1f8a25)